lib/azure/armrest/configuration.rb

```diff
@@ -124,7 +124,7 @@ def initialize(args)
         # Delay this to avoid a double call
         @environment = options.delete(:environment)
 
-        if @environment.to_s.casecmp(Azure::Armrest::USGOV_ENVIRONMENT) == 0
+        if @environment.to_s.downcase == Azure::Armrest::USGOV_ENVIRONMENT
```

Comment:
Forgive me for not thinking thoroughly. The above suggestion is still buggy. We need to think the following combination:

1. no urls, no environment (the most common one)
2. no urls, but with environment (such as usgov)
3. with urls, no environment (user case?)
4. with urls and environment (do we allow because they may conflict?)